src/inmanta/ast/statements/define.py

```diff
@@ -358,7 +324,11 @@ def __init__(
         self.anchors = [TypeReferenceAnchor(x.namespace, x) for x in implementations]
         self.anchors.append(TypeReferenceAnchor(entity_name.namespace, entity_name))
         self.anchors.extend(select.get_anchors())
+        self.location = entity_name.get_location()
+        if inherit and (not isinstance(select, Literal) or select.value is not True):
+            raise RuntimeException(self, "Conditional implementation with parents not allowed")
```

Comment:
Is this OK or should it be caught by syntax?

Reply:
ok